Robot.java

```diff
@@ -0,0 +1,108 @@
+import edu.wpi.first.wpilibj.DoubleSolenoid;
+import edu.wpi.first.wpilibj.TimedRobot;
+import edu.wpi.first.wpilibj.VictorSPX;
+import edu.wpi.first.wpilibj.XboxController;
+
+/**
+* The VM is configured to automatically run this class, and to call the functions corresponding to
+* each mode, as described in the TimedRobot documentation. If you change the name of this class or
+* the package after creating this project, you must also update the build.gradle file in the
+* project.
+*/
+public class Robot extends TimedRobot {
+    private VictorSPX mMotor;
+    private XboxController mController;
+    private Compressor mCompressor;
+    private DoubleSolenoid mSolenoidOne;
+    private DoubleSolenoid mSolenoidTwo;
+    private int loopCounter;
+
+    /**
+    * This function is run when the robot is first started up and should be used for any
+    * initialization code.
+    */
+    @Override
+    public void robotInit() {
+        mMotor = new VictorSPX(2);
+        mController = new XboxController(0);
+        mCompressor = new Compressor(0);
+        mCompressor.setClosedLoopControl(true);
+    }
+
+    /**
+    * This function is called every robot packet, no matter the mode. Use this for items like
+    * diagnostics that you want ran during disabled, autonomous, teleoperated and test.
+    *
+    * <p>This runs after the mode specific periodic functions, but before LiveWindow and
+    * SmartDashboard integrated updating.
+    */
+    @Override
+    public void robotPeriodic() {}
+
+    /**
+    * This autonomous (along with the chooser code above) shows how to select between different
+    * autonomous modes using the dashboard. The sendable chooser code works with the Java
+    * SmartDashboard. If you prefer the LabVIEW Dashboard, remove all of the chooser code and
+    * uncomment the getString line to get the auto name from the text box below the Gyro
+    *
+    * <p>You can add additional auto modes by adding additional comparisons to the switch structure
+    * below with additional strings. If using the SendableChooser make sure to add them to the
+    * chooser code above as well.
+    */
+    @Override
+    public void autonomousInit() {
+        loopCounter = 0;
+    }
+
+    /** This function is called periodically during autonomous. */
+    @Override
+    public void autonomousPeriodic() {
+        if (loopCounter < 18) {
+            mSolenoidTwo.set(DoubleSolenoid.Value.kReverse);
+            mSolenoidTwo.set(DoubleSolenoid.Value.kForward);
+
+            loopCounter++;
+        }
+    }
+
+    /** This function is called once when teleop is enabled. */
+    @Override
+    public void teleopInit() {}
+
+    /**
+    * This function is called periodically during operator control. If a certain key is pressed on
+    * the joystick,
+    */
+    @Override
+    public void teleopPeriodic() {
+        mSolenoidOne = new DoubleSolenoid(1, 2);
+        mSolenoidOne = new DoubleSolenoid(1, 1, 2);
```

Comment:
Please do these initializations in the `robotInit()` method. The `teleopPeriodic()` method runs periodically and we don't want to re initialize the Solenoids at every loop (that will most likely crash the program).